### PR TITLE
debug: allow configuring variable value length limit

### PR DESCRIPTION
**Pull request by oh-summy**
### Why are the changes in this PR needed?

The DAP debugger currently truncates variable values to a hardcoded
limit of 100 characters, with no way for callers to configure it.
Long values cannot be inspected or copied whole from a debugger UI.

### What are the changes in this PR?

- Add a new `SetMaxVariableLength` Debugger option (in both
  `v1/debug` and the top-level `debug` package)
- A value of 0 disables truncation; the default stays 100 characters
  for backward compatibility
- Plumb the limit through `variableManager` and `namedVar`, so it
  applies to top-level and nested (object/array/set) variables alike
- Relax `truncatedString` to return the original string unchanged
  when the limit is <= 0
- Add `TestTruncatedString` and `TestVariableValueLengthLimit`
  covering the default limit, unlimited (0), negative, custom limits,
  and boundary cases

### Notes

- `go build ./...` and `go test ./v1/debug/...` both pass
- Default behavior is unchanged; the new option is opt-in
- This PR was developed with AI assistance (Claude Code)

### Further comments

Refs #8890

### CODE REVIEW

File: v1/debug/variable.go [hunk 1]
```diff
@@ -107,6 +110,9 @@ func (vs *variableManager) vars(varRef VarRef) ([]Variable, error) {
 }
 
 func truncatedString(s string, max int) string {
+	if max <= 0 {
```
Comment (sspaink) [inline v1/debug/variable.go]: If max is 1 this would panic when it tries `s[:max-2]`, or if it is 2 it would return just `...`.

maybe it should be:

```suggestion
	if max <= 3 {
```

Comment (oh-summy) [reply v1/debug/variable.go]: Good catch — changed to `max <= 3` as suggested, and added test cases for max=1, 2, 3 to prevent regression.

File: v1/debug/variable_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
+// Copyright 2024 The OPA Authors.  All rights reserved.
```
Comment (sspaink) [inline v1/debug/variable_test.go]:
```suggestion
// Copyright 2026 The OPA Authors.  All rights reserved.
```

File: v1/debug/variable.go [hunk 3]
```diff
@@ -141,8 +147,9 @@ func (vs *variableManager) subVars(v ast.Value) VarRef {
 		vars := make([]namedVar, 0, obj.Len())
 		if err := obj.Iter(func(k, v *ast.Term) error {
 			vars = append(vars, namedVar{
-				name:  k.String(),
-				value: v.Value,
+				name:              k.String(),
+				value:             v.Value,
+				maxVariableLength: vs.maxVariableLength,
```
Comment (sspaink) [inline v1/debug/variable.go]: `maxVariableLength` is already getting set in the `vars` method (line 98), redundant to also set it here. Same applies to line 169 and 182. since all getters' vars flow through vars().

**Review by sspaink — CHANGES_REQUESTED** [review]

File: v1/debug/variable.go (hunk 3 above)
Comment (oh-summy) [reply v1/debug/variable.go]: Thanks! All fixed:

- Copyright year → 2026
- Removed redundant maxVariableLength assignments in subVars (vars() already does it)

Ready when you are

File: v1/debug/variable_test.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/open-policy-agent/opa/v1/ast"
+)
+
+func TestTruncatedString(t *testing.T) {
+	t.Helper()
```
Comment (sspaink) [inline v1/debug/variable_test.go]: `t.Helper()` doesn't do anything called from a top-level test function. It is meant to be used in helper functions invoked by a test. Can you remove this and the one in `TestVariableValueLengthLimit` ?

Comment (oh-summy) [reply v1/debug/variable_test.go]: Done — removed t.Helper() from both TestTruncatedString and TestVariableValueLengthLimit. Let me know if anything else is needed.

**Review by sspaink — APPROVED** [review]
Thank you!
